[28기 이가윤] 메인 페이지 구현완료

Gayun00 opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [ ] 리뷰 반영
- [ ] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

[Add] subSlider 레이아웃  구현
 - 목업데이터 입력
 - css 진행중

[Modify] mainslider 수정

[Modify] subslider 수정

[Modify] 너비 고정

[Modify] css 수정 중



## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
- 아이디, 비밀번호 입력창을 사용한 로그인 기능 구현 (예시)

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)


<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)


<br />

## :: 기타 질문 및 특이 사항
subSliderItem 컴포넌트가 화면에 표시되지 않습니다.
package.json, package-lock.json파일 깃 충돌 수정 후 일어난 오류라 뭔가 잘못 건드린 것 같습니다.
props가 잘 전달되었는지는 확인했는데 slick라이브러리에 오류가 생겨 표시가 되지 않는 것 같습니다. 어떻게 해야할지 잘 모르겠습니다.

bonbon0202 commented on public/data/subslider.json:
```diff
@@ -0,0 +1,37 @@
+        "img": "",
+        "name": "베로마",
+        "location": "부산/해운대구",
+        "price": "2000~25000"
```
price range 이죠?! 아마 백엔드에서는 가격을 string이 아닌 minPrice, maxPrice 등의 key 값으로 Int (number)  타입으로 관리할 것 같은데 확인해주세요!

bonbon0202 commented on src/pages/Main/Main.js:
```diff
@@ -1,7 +1,27 @@
 
 function Main() {
-  return <div>main</div>;
+  const [mainSlide, setMainSlide] = useState([]);
+  useEffect(() => {
+    fetch('/data/mainSlider.json')
+      .then(res => res.json())
+      .then(res => setMainSlide([...res]));
+  }, []);
+  return (
```
가독성을 위해 useEffect 기준으로 위 아래, 한 줄씩만 띄어주세요-!

bonbon0202 commented on src/pages/Main/MainSlider.js:
```diff
@@ -0,0 +1,146 @@
+
+  img {
+    object-fit: cover;
+    /* margin: 0 5rem 0 5rem; */
```
불필요한 주석이면 삭제해주세요!

bonbon0202 commented on src/pages/Main/MainSlider.js:
```diff
@@ -0,0 +1,146 @@
+
+const ButtonWrapper = styled.div`
+  position: absolute;
+  top: 69%;
```
top에서 69%이면 bottom을 기준으로 하는 것은 어떨까요?! 디스플레이의 크기에 따라 오차범위가 클 것 같네요!

bonbon0202 commented on src/pages/Main/SubSlider/SubSliderItem.js:
```diff
@@ -0,0 +1,67 @@
+
+function SubSliderItem() {
+  return (
+    <Wrapper>
```
- styled-component를 이용해서 스타일링이 된 태그들은 해당 definition으로 이동해서 바로 스타일링을 확인이 가능한데 className으로 선언한 부분은 바로 definition으로 이동이 불가능해서 스타일링 파악하기에 어려움이 있습니다.
- 위의 내용에 덧붙여서 className을 이용한 스타일링의 nesting depth가 깊어지는 경우에는 더 더욱 찾기가 힘들어집니다
- depth가 깊어지는 경우 뿐만 아니라 1depth로도 className을 이용한 스타일 선언부가 길어지는 경우에도 해당 스타일링을 찾기 어려워집니다.
- 위의 사항들로 인해서 CSS in JS(styled-components 등)을 이용한 스타일링을 하는 경우에는 라이브러리 사용상 강제되는 등의 불가피한 부분을 제외하고는 className을 통한 스타일링은 최대한 지양하는게 권장됩니다.

bonbon0202 reviewed (commented):
가윤님! slick 이용하셔서 슬라이드 구현해주셨군요! 
잘 해주셨는데 중간중간 참조하실 부분에 대한 코멘트 드렸습니다! 일단, 불필요한 주석처리도 모두 지워주시고 다시 한 번 더 push 해주세요!

ghost commented:
✅ Sider has resolved the errors or warnings via the commit de8962f! See the [analysis result](https://sider.review/gh/repos/446331007/pulls/6) for details.

---

You can [turn off such notifications](https://sider.review/gh/repos/446331007/settings#notifications) if unnecessary.

bonbon0202 commented on src/pages/Main/MainSlider.js:
```diff
@@ -5,79 +5,69 @@ import 'slick-carousel/slick/slick.css';
-    );
-  }
+function MainSlider({ mainSlide }) {
+  // function ArrowLeft(props) {
+  //   return (
+  //     <button {...props} className="prev">
+  //       <IoIosArrowBack className="arrowBack" />
+  //     </button>
+  //   );
+  // }
+  // function ArrowRight(props) {
+  //   return (
+  //     <button {...props} className="next">
+  //       <IoIosArrowForward className="arrowForward" />
+  //     </button>
+  //   );
+  // }
```
불필요하신 주석이라면 모두 삭제해주세요!

bonbon0202 commented on src/pages/Main/SubSlider/SubSlider.js:
```diff
@@ -1,77 +1,98 @@
+      <SliderTitle>{subSlide[0].title}</SliderTitle>
+      <SliderWrapper>
+        <Slider {...settings}>
+          {subSlide[0].hotels.map((hotel, idx) => (
```
보통은 데이터에 접근할 때, 인덱스로 접근하는 경우는 불안정하기 때문에 이런 경우는 객체 안에 배열로 있는 데이터 구조가 더 적합할 것 같네요-! 참조만해주세요.

Gayun00 replied:
넵!

bonbon0202 commented on src/pages/Main/SubSlider/SubSlider.js:
```diff
@@ -1,77 +1,98 @@
+  font-size: 1.5rem;
+`;
+
+const SliderWrapper = styled.div``;
```
style이 불필요한 요소는 컴포넌트가 아닌 그냥 '요소'로 적어주셔도 됩니다.

Gayun00 replied:
넵 알겠습니다!

bonbon0202 reviewed (commented):
가윤님! 불필요한 주석이 너무 많이 보이는 것 같습니다. 
주석만 삭제해주시고 다른 리뷰는 참조해주세요.

Gayun00 committed:
[Fix] recoil 재설치

[Add] 어드민 페이지 레이아웃 추가

[Add] 이미지 업로드 값 인코딩 기능 구현

[Refactor] 리팩토링 중

[Add] 가격 변경, 호텔 삭제 기능 추가

[Fix]깃 충돌 수정

[Fix] recoil 재설치

[Fix] reset

[Fix] package.json 충돌 수정

[Fix] 충돌 수정 중
